src/microsoft-extensions-ai/abstraction-implementations/AbstractionImplementationExamples/DependencyInjection.cs

```diff
@@ -0,0 +1,34 @@
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+public partial class AbstractionSamples
+{
+    public static async Task DependencyInjection() 
+    {
+        var app = Host.CreateApplicationBuilder();
+
+        app.Services.AddSingleton<IDistributedCache, InMemoryCacheStorage>();
```

Comment:
What is InMemoryCacheStorage and why do we need it? Can this instead just be `app.Services.AddDistributedMemoryCache()`?  That comes from the Microsoft.Extensions.Caching.Memory package.

Reply:
Can you point me to a sample thatshows how to use the in-memory cache from Microsoft.Extensions.Caching.Memory package? I didn't see that in the MEAI repo and only found this custom memory cache implementation.

Reply:
Literally replace:
```
app.Services.AddSingleton<IDistributedCache, InMemoryCacheStorage>();
```
with
```
app.Services.AddDistributedMemoryCache()
```

Reply:
That works if you have a DI container. I found the version that works without one. Thanks.

Reply:
Right, this example has a DI container :)